[feature]: feedback form

0x009922 opened the pull request.

outoftardis commented on .vitepress/theme/components/ShareFeedback.vue:
```diff
@@ -0,0 +1,193 @@
+            <textarea
+              id="feedback-input-text"
+              v-model="feedbackText"
+              placeholder="What we can do to improve the overall documentation browsing experience?"
```
```suggestion
              placeholder="What can we do to improve the overall documentation browsing experience?"
```

outoftardis commented on .vitepress/theme/components/ShareFeedback.vue:
```diff
@@ -0,0 +1,193 @@
+          </div>
+
+          <div>
+            <label for="feedback-input-contact"><i>(optional)</i> Contact address</label>
+
+            <input
+              id="feedback-input-contact"
+              v-model="contact"
+              placeholder="Email or any other contact address"
+            >
```
```suggestion
            <label for="feedback-input-contact"><i>(optional)</i> Contact information</label>

            <input
              id="feedback-input-contact"
              v-model="contact"
              placeholder="Email address or any other way to contact you (Discord, Telegram)"
            >
```

outoftardis reviewed (commented):
appearance-wise: I think it'd be better to put the "provide feedback" button on the right side of the page (if that's possible)

0x009922 commented:
> appearance-wise: I think it'd be better to put the "provide feedback" button on the right side of the page (if that's possible)

@outoftardis, it is possible to put it in the place where "On this page" section is rendered. The problem is that "left aside bar" is only rendered when the page is full-width. Thus, it becomes impossible to share feedback from a mobile device.

6r1d commented:
As you may have seen in my version, https://github.com/hyperledger/iroha-2-docs/pull/248, there was a feedback type. There are two key types, a **bug** and a **suggestion**. In our admin panel, we'll see both marked with icons and will be able to address bugs faster. A bug essentially means a user can't replicate a behaviour we intended Iroha 2 to have, and I believe it's important to distinguish the two. What do you think?

6r1d commented:
Notification about an error could be simplified as well. Also, instead of an error, we can suggest filling a GitHub issue for the docs.

outoftardis commented:
@6r1d I think we definitely need to distinguish between bugs and suggestions. We could offer the option to file an issue on Github but we probably shouldn't limit the options, i.e. the users would be able to choose between describing the issue briefly in a feedback form vs. filing an issue with more details.

I think we can also have a smaller "report a bug" button on the page (e.g. in top right corner next to github logo) that leads directly to the doc issue template.

6r1d commented:
> I think we can also have a smaller "report a bug" button on the page (e.g. in the top right corner next to the GitHub logo) that leads directly to the doc issue template.
That sounds very good

> We could offer the option to file an issue on Github
I see it only as a backup plan: users who want to file a detailed issue would know where to look quickly.
Others would need more help, and we expect the reports to be more fragmented; in that case, we make an issue after seeing initial comments and (maybe) asking questions.

0x009922 commented:
- Added **Bug**, **Suggestion** and **Other** feedback types. 
- Added a bug icon near the GitHub one that points to the issue creating page
- Swapped placeholder/label for contact field - placeholder was too long
- Fixed several ui issues

@outoftardis @6r1d

outoftardis commented on .vitepress/theme/components/ShareFeedback.vue:
```diff
@@ -0,0 +1,252 @@
+            <textarea
+              id="feedback-input-text"
+              v-model="feedbackText"
+              placeholder="What can we do to improve the overall documentation browsing experience?"
```
Do we need this placeholder? It doesn't really work with the bug type

0x009922 replied:
We don't really need it, but some placeholder is a good practice. I can make it dynamic depending on what option is selected, btw.

outoftardis replied:
Dynamic is a good idea.

0x009922 replied:
Okay. What text should I use?

outoftardis replied:
for bugs:
> Report any bugs or issues you found in Iroha 2 documentation.

0x009922 replied:
Done.

outoftardis commented on .vitepress/theme/components/ShareFeedback.vue:
```diff
@@ -0,0 +1,252 @@
+          <div>
+            <fieldset class="space-y-1">
+              <legend class="field-label">
+                What kind of feedback?*
```
```suggestion
                Feedback type*
```

outoftardis commented on .vitepress/theme/components/ShareFeedback.vue:
```diff
@@ -0,0 +1,252 @@
+              for="feedback-input-contact"
+              class="field-label"
+            >
+              <i>(optional)</i> Email address or any other way to contact you (Discord, Telegram)
+            </label>
+
+            <input
+              id="feedback-input-contact"
+              v-model="contact"
+              placeholder="Contact information"
```
this might be better

```suggestion
              <i>(optional)</i> Contact information
            </label>

            <input
              id="feedback-input-contact"
              v-model="contact"
              placeholder="Email address, Discord, or Telegram"
```

6r1d reviewed (approved):
LGTM
